echopype/mask/api.py

```diff
@@ -0,0 +1,306 @@
+    freqAB: Optional[List[float]] = None,
+    chanAB: Optional[List[str]] = None,
+    operator: str = ">",
+    diff: Union[float, int] = None,
+) -> xr.DataArray:
+    """
+    Create a mask based on the differences of Sv values using a pair of
+    frequencies. This method is often referred to as the "frequency-differencing"
+    or "dB-differencing" method.
+
+    Parameters
+    ----------
+    source_Sv: xr.Dataset or str or pathlib.Path
+        If a Dataset this value contains the Sv data to create a mask for,
+        else it specifies the path to a zarr or netcdf file containing
+        a Dataset. This input must correspond to a Dataset that has the
+        coordinate ``channel`` and variables ``frequency_nominal`` and ``Sv``.
+    storage_options: dict, optional
+        Any additional parameters for the storage backend, corresponding to the
+        path provided for ``source_Sv``
+    freqAB: list of float, optional
+        The pair of nominal frequencies to be used for frequency-differencing, where
+        the first element corresponds to ``freqA`` and the second element corresponds
+        to ``freqB`` (see notes below for more details)
+    chanAB: list of float, optional
+        The pair of channels that will be used to select the nominal frequencies to be
+        used for frequency-differencing, where the first element corresponds to ``freqA``
+        and the second element corresponds to ``freqB`` (see notes below for more details)
+    operator: {">", "<", "<=", ">=", "=="}
+        The operator for the frequency-differencing
+    diff: float or int
+        The threshold of Sv difference between frequencies
+
+    Returns
+    -------
+    xr.DataArray
+        A DataArray containing the mask for the Sv data. Regions satisfying the thresholding
+        criteria are filled with ``True``, else the regions are filled with ``False``.
+
+    Raises
+    ------
+    RuntimeError
+        If neither ``freqAB`` or ``chanAB`` are given
+    RuntimeError
+        If both ``freqAB`` and ``chanAB`` are given
+    TypeError
+        If any input is not of the correct type
+    RuntimeError
+        If either ``freqAB`` or ``chanAB`` are provided and the list
+        does not contain 2 distinct elements
+    RuntimeError
+        If ``freqAB`` contains values that are not contained in ``frequency_nominal``
+    RuntimeError
+        If ``chanAB`` contains values that not contained in ``channel``
+    RuntimeError
+        If ``operator`` is not one of the following: ``">", "<", "<=", ">=", "=="``
+    RuntimeError
+        If the path provided for ``source_Sv`` is not a valid path
+    RuntimeError
+        If ``freqAB`` or ``chanAB`` is provided and the Dataset produced by ``source_Sv``
+        does not contain the coordinate ``channel`` and variable ``frequency_nominal``
+
+    Notes
+    -----
+    Either ``freqAB`` or ``chanAB`` must be provided, but both parameters
+    cannot be given.
```

Comment:
```suggestion
```
I suggest removing these and added description upfront under both argin description so that it is read in one place.